### PR TITLE
3.30.0: Adds new SDK versions and contract files

### PR DESCRIPTION
**Pull request by kundadebdatta**
# Pull Request Template

## Description

This bumps the SDK version from 3.29.0 to 3.30.0. Please see the below changes in the GA public contract.

contract_diff_3_29_0_and_3_30_0_aug_19 
Also highlighted below are the preview contract changes.

contract_diff_3_29_0_preview_and_3_30_0_preview_aug_19

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [#3197](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3197) Change Feed: Adds SDK changes required for Full-Fidelity Change Feed
+
+### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-17
+
+#### Added
+- [#3376](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3376) Client Telemetry : Refactors code to compute hash of VM ID and Process Name information
+- [#3364](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3364) Integrated cache: Adds DedicatedGatewayRequestOptions for public release
+- [#3361](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3361) Documentation: Adds Contributing guide
+- [#3353](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3353) Contacted Regions Added to TraceSummary
+- [#3273](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3273) Linq: Adds support constant evaluation of Nullable<T>.HasValue
```
Comment (ealsur) [inline changelog.md:29]:
```suggestion
- [#3273](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3273) Linq: Adds support constant evaluation of Nullable<T>.HasValue. Thanks (Thanks [@ccurrens](https://github.com/ccurrens))
```

File: changelog.md [hunk 2]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [#3331](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3331) AIIntegration: Adds Client and other information in attributes
+- [#3197](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3197) Change Feed: Adds SDK changes required for Full-Fidelity Change Feed
+
+### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-17
+
+#### Added
+- [#3376](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3376) Client Telemetry : Refactors code to compute hash of VM ID and Process Name information
+- [#3364](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3364) Integrated cache: Adds DedicatedGatewayRequestOptions for public release
+- [#3361](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3361) Documentation: Adds Contributing guide
+- [#3353](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3353) Contacted Regions Added to TraceSummary
```
Comment (ealsur) [inline changelog.md:28]: Remove this entry, it is internal

File: changelog.md [hunk 3]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+#### Added
+- [#3331](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3331) AIIntegration: Adds Client and other information in attributes
+- [#3197](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3197) Change Feed: Adds SDK changes required for Full-Fidelity Change Feed
+
+### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-17
+
+#### Added
+- [#3376](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3376) Client Telemetry : Refactors code to compute hash of VM ID and Process Name information
+- [#3364](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3364) Integrated cache: Adds DedicatedGatewayRequestOptions for public release
+- [#3361](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3361) Documentation: Adds Contributing guide
```
Comment (ealsur) [inline changelog.md:27]: Remove this entry, not a change in the SDK, just a doc update on the repo

File: changelog.md [hunk 4]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### <a name="3.30.0-preview"/> [3.30.0-preview](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0-preview) - 2022-08-17
+
+#### Added
+- [#3331](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3331) AIIntegration: Adds Client and other information in attributes
```
Comment (ealsur) [inline changelog.md:19]:
```suggestion
- [#3331](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3331) Open Telemetry: Adds Client and other information in attributes
```

File: changelog.md [hunk 5]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-17
+
+#### Added
+- [#3376](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3376) Client Telemetry : Refactors code to compute hash of VM ID and Process Name information
+- [#3364](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3364) Integrated cache: Adds DedicatedGatewayRequestOptions for public release
+- [#3361](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3361) Documentation: Adds Contributing guide
+- [#3353](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3353) Contacted Regions Added to TraceSummary
+- [#3273](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3273) Linq: Adds support constant evaluation of Nullable<T>.HasValue
+- [#3268](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3268) Diagnostics: Adds GetStartTimeUtc and GetFailedRequestCount
+- [#3241](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3241) Client Encryption: Adds code to support PartitionKey and Id encryption.
```
Comment (ealsur) [inline changelog.md:31]: Remove, this is for the Encryption project, not relevant here

**Review by ealsur — COMMENTED** [review]
Please address the comments

**Comment by kirankumarkolli** [comment]
- For TTL expired documents what's expected on ChangeFeedMetadata? (ex: OperationType will be delete and TTLExpire=true). Please cover it code documentaion. 
- TimeToLiveExpired: How about 'IsTimeToLiveExpired'?

File: changelog.md [hunk 6]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
...
+
+#### Fixed
+- [#3350](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3350) Diagnostics: Fixes Diagnostics for Query with FeedRanges
+- [#3348](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3348) Documentation: Fixes DeleteItemAsync Example
```
Comment (kirankumarkolli) [inline changelog.md:35]: Are these code documents?

File: changelog.md (hunk 1 above)
Comment (bartelink) [reply changelog.md:29]: thanks x2 typo ;) @ealsur

File: changelog.md [hunk 7]
```diff
@@ -13,6 +13,29 @@ Preview features are treated as a separate branch and will not be included in th
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### <a name="3.30.0-preview"/> [3.30.0-preview](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0-preview) - 2022-08-17
+
+#### Added
+- [#3331](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3331) AIIntegration: Adds Client and other information in attributes
+- [#3197](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3197) Change Feed: Adds SDK changes required for Full-Fidelity Change Feed
+
+### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-17
```
[the remark this answers is not in the document]
Comment (kundadebdatta) [reply changelog.md]: Done.

File: changelog.md (hunk 1 above)
Comment (kundadebdatta) [reply changelog.md]: Thank for bringing this up! Added.

File: changelog.md (hunk 2 above)
Comment (kundadebdatta) [reply changelog.md]: Done.

File: changelog.md (hunk 3 above)
Comment (kundadebdatta) [reply changelog.md]: Done.

File: changelog.md (hunk 4 above)
Comment (kundadebdatta) [reply changelog.md]: Updated as per suggestion.

File: changelog.md (hunk 5 above)
Comment (kundadebdatta) [reply changelog.md]: Removed.

File: changelog.md [hunk 8]
```diff
@@ -13,22 +13,19 @@ Preview features are treated as a separate branch and will not be included in th
...
-### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-17
+### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-19
 
 #### Added
 - [#3376](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3376) Client Telemetry : Refactors code to compute hash of VM ID and Process Name information
 - [#3364](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3364) Integrated cache: Adds DedicatedGatewayRequestOptions for public release
-- [#3361](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3361) Documentation: Adds Contributing guide
-- [#3353](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3353) Contacted Regions Added to TraceSummary
-- [#3273](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3273) Linq: Adds support constant evaluation of Nullable<T>.HasValue
+- [#3273](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3273) Linq: Adds support constant evaluation of Nullable<T>.HasValue. Thanks (Thanks [@ccurrens](https://github.com/ccurrens))
```
Comment (bartelink) [inline changelog.md:27]:
```suggestion
- [#3273](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3273) Linq: Adds support constant evaluation of `Nullable<T>.HasValue`. (Thanks [@ccurrens](https://github.com/ccurrens))
```

Comment (kundadebdatta) [reply changelog.md]: Thanks @bartelink !

File: changelog.md (hunk 6 above)
Comment (kundadebdatta) [reply changelog.md:32]: That is correct. The documentation belongs to the code.

**Comment by philipthomas-MSFT** [comment]
>

**Comment by philipthomas-MSFT** [comment]
> * For TTL expired documents what's expected on ChangeFeedMetadata? (ex: OperationType will be delete and TTLExpire=true). Please cover it code documentaion.
> * TimeToLiveExpired: How about 'IsTimeToLiveExpired'?

@kirankumarkolli 
AFAIK, yes, if the OperationType is 'deleted' then the TimeToLiveExpired will be 'true'. 
Also, are you suggesting a refactor to change TimeToLiveExpired to 'IsTimeToLiveExpired'?

File: changelog.md [hunk 9]
```diff
@@ -13,6 +13,26 @@ Preview features are treated as a separate branch and will not be included in th
...
+### <a name="3.30.0"/> [3.30.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.30.0) - 2022-08-19
+
+#### Added
+- [#3376](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3376) Client Telemetry : Refactors code to compute hash of VM ID and Process Name information
+- [#3364](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3364) Integrated cache: Adds DedicatedGatewayRequestOptions for public release
+- [#3273](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3273) Linq: Adds support constant evaluation of `Nullable<T>.HasValue`. (Thanks [@ccurrens](https://github.com/ccurrens))
+- [#3268](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3268) Diagnostics: Adds GetStartTimeUtc and GetFailedRequestCount
+
+#### Fixed
+- [#3350](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3350) Diagnostics: Fixes Diagnostics for Query with FeedRanges
```
Comment (ealsur) [inline changelog.md:31]: Please add https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3391

Comment (kundadebdatta) [reply changelog.md:31]: Let me add it.

Comment (kundadebdatta) [reply changelog.md:32]: Done!

**Comment by kundadebdatta** [comment]
> 

I have updated the contract to reflect the correct naming convention.

**Review by ealsur — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**